packages/node/src/executors/execute/execute.impl.spec.ts

```diff
@@ -329,4 +342,230 @@ describe('NodeExecuteBuilder', () => {
       }
     });
   });
+
+  describe('--watch', () => {
+    let mockSubProcessExit;
+    let mockSubProcessMessage;
+    let mockSubProcessError;
+
+    beforeEach(() => {
+      fork.mockImplementation(() => {
+        mockSubProcess = {
+          on: jest.fn().mockImplementation((eventName, cb) => {
+            if (eventName === 'exit') {
+              mockSubProcessExit = (exitCode: number) => {
+                cb(exitCode);
+                mockSubProcess.exitCode = exitCode;
+              };
+            }
+
+            if (eventName === 'error') {
+              mockSubProcessError = cb;
+            }
+
+            if (eventName === 'message') {
+              mockSubProcessMessage = cb;
+            }
+          }),
+          exitCode: null,
+        };
+
+        return mockSubProcess;
+      });
+    });
+
+    describe('false', () => {
+      it('Should not complete until the child process has exited', async () => {
+        const firstEvent = Symbol('first');
+        const secondEvent = Symbol('second');
+        const events = [];
+        const executorIterator = executeExecutor(
+          {
+            ...testOptions,
+            watch: false,
+          },
+          context
+        );
+
+        expect(await executorIterator.next()).toEqual({
+          done: false,
+          value: {
+            outfile: 'outfile.js',
+            success: true,
+          },
+        });
+
+        const result = await Promise.race([
+          executorIterator.next(),
+          new Promise(() => {
+            mockSubProcessExit(1);
+            events.push(firstEvent);
+          }),
+        ]);
```

Comment:
```suggestion
        const result = await Promise.race([
          executorIterator.next(),
          new Promise(() => {
            mockSubProcessExit(1);
            events.push(firstEvent);
            // NOTE: this promise intentionally never resolves so that...
          }),
        ]);
```

This threw me off for a sec :) Might help to leave some breadcrumbs